src/App.js

```diff
@@ -1,19 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const [likeCount, setLikeCount] = useState(0);
```

Comment:
👀 The number of liked messages can be determined from inspecting the current list of messages. Prefer to avoid introducing additional state when the existing state provides the information we need, and without the risk of state getting out of synch.

We can use a few approaches to get the liked count from the message data. The most javascript-esque way is probably to use `reduce`, like
```js
  const likeCount = messages.reduce((total, message) => {
    return total + message.liked;  // true will act as 1, false as 0
  }, 0)
```

Another approach (which makes a copy of the list, but which doesn't really matter since we're `map`ping this data later anyway) would be
```js
  const likeCount = messages.filter(message => message.liked).length;
```

Either approach does need to iterate over the message list, which we can avoid with the stand alone count approach. However, since we map over the list later anyway, the additional cost of the iteration here is negligible. If we did end up with other interactions that caused a re-render and didn't want to perform the calculation unnecessarily, there are ways to memoize results in React that can be used to avoid unnecessary calculations. But we should start by deriving values from existing state when possible, and only look for optimizations if the rendering performance is adversely impacted.